### PR TITLE
Suppress error code if copying of README.md/amalgamate_demo.cpp isn't needed

### PR DESCRIPTION
**Pull request by pps83**
Fixes an error introduced by https://github.com/simdjson/simdjson/pull/1070

### CODE REVIEW

File: singleheader/amalgamate.sh [hunk 1]
```diff
@@ -127,8 +127,8 @@ echo "/* auto-generated on ${timestamp}. Do not edit! */" > ${AMAL_C}
     done
 } >> ${AMAL_C}
 
-cp -u "${SCRIPTPATH}/amalgamate_demo.cpp" "${DEMOCPP}" 2>/dev/null
-cp -u "${SCRIPTPATH}/README.md" "${README}" 2>/dev/null
+cp -u "${SCRIPTPATH}/amalgamate_demo.cpp" "${DEMOCPP}" 2>/dev/null || :
```
Comment (jkeiser) [inline singleheader/amalgamate.sh:130]: What does the `:` do? This is new syntax to me.

Will cp -f do what we want? I don't care which, as long as this works everywhere.

Comment (jkeiser) [reply singleheader/amalgamate.sh:130]: Also, if the source file isn't there and it therefore fails to copy, will we still get an error code?

Comment (lemire) [reply singleheader/amalgamate.sh:130]: The "-u" flag is GNU-specific. It is likely not to work under freeBSD and macos.

**Comment by lemire** [comment]
Under non-GNU systems (so FreeBSD and macOS)...

```
AMALGAMATE_OUTPUT_PATH=. ../path/amalgamate.sh
```

will only copy simdjson.cpp and simdjson.h.

**Comment by lemire** [comment]
Please don't use GNU-specific flags. It is bad practice.

**Comment by lemire** [comment]
Currently the build is broken if you have bash but are not under a GNU system (essentially Linux and the like).


We cannot use `-u`. Please keep the script portable.

See PR:
https://github.com/simdjson/simdjson/pull/1109

**Comment by lemire** [comment]
I'll close this PR. If there is still an issue, please generate a new PR.

**Comment by pps83** [comment]
Same fix still needs to be applied regardless of #1109, see updated PR:
https://github.com/simdjson/simdjson/pull/1113

**Pull request closed without merging**